app/services/catalogue_services_client.rb

```diff
@@ -25,6 +27,46 @@ def get_holdings(instance_id:)
     end
   end
 
+  def get_holding(instance_id:, holdings_id:, item_id:)
+    all_holdings = get_holdings(instance_id: instance_id)
+
+    # find holdings record
+    holding = all_holdings.select { |h| h["id"] == holdings_id }.first
+
+    # find item record
+    item = holding["itemRecords"].select { |i| i["id"] == item_id }.first
+
+    [holding, item]
+  end
+
+  def create_request(requester:, request:)
+    set_token
+    conn = Faraday.new(url: ENV["CATALOGUE_SERVICES_API_BASE_URL"]) do |f|
+      f.request :authorization, "Bearer", @bearer_token
+      f.response :json
+    end
+
+    request_body = JSON.generate(requesterId: requester,
+      instanceId: request[:instance_id],
+      holdingsId: request[:holdings_id],
+      itemId: request[:item_id],
+      year: [request[:year]],
+      enumeration: request[:enumeration],
+      chronology: request[:chronology],
+      barcode: request[:barcode],
```

Comment:
Note that barcode will be ignored for checkouts, it's generated by catalogue-services for serials and taken from Folio item metadata for monographs.

Reply:
Removed the `barcode`.